### PR TITLE
added password dialog for encrypted templates

### PR DESCRIPTION
**Pull request by rafapaezbas**

### CODE REVIEW

File: cmd/init.js [hunk 1]
```diff
@@ -39,12 +39,22 @@ module.exports = (ipc) => async function init (cmd) {
...
   } catch (err) {
-    if (err.code !== 'ERR_PERMISSION_REQUIRED') throw err
-    const explain = 'Be sure that software is trusted before using it\n' +
-    '\nType "TRUST" to allow template initialization or anything else to exit\n\n'
-    const ask = 'Trust template'
-    const act = 'Use pear init again to initalize from trusted template'
-    await trust({ ipc, key: err.info.key, message: err.message, explain, ask, act })
+    if (err.code !== 'ERR_PERMISSION_REQUIRED') {
+      throw err
+    } else {
```
Comment (davidmarkclements) [inline cmd/init.js:44]: if you throw you exit the function - no need for the else branch - also means you can de-nest 
```
if (err.code !== 'ERR_PERMISSION_REQUIRED') throw err
//rest of code at this indent
```

File: cmd/init.js [hunk 2]
```diff
@@ -39,12 +39,22 @@ module.exports = (ipc) => async function init (cmd) {
...
+        const message = 'Added encryption key, run init again to complete it.'
+        await password({ ipc, key: err.info.key, explain, message })
+      } else {
+        const explain = 'Be sure that software is trusted before using it\n' +
+          '\nType "TRUST" to allow template initialization or anything else to exit\n\n'
+        const ask = 'Trust template'
+        const act = 'Use pear init again to initalize from trusted template'
+        await trust({ ipc, key: err.info.key, message: err.message, explain, ask, act })
+      }
+    }
```
Comment (davidmarkclements) [inline cmd/init.js:57]: this is what iface `permit` should actually do 

this code from 45-57 here should just be one line

```
await permit({ ipc, info: err.info, message: err.message })
```

and then permit should contain the conditional logic to then call `trust` / `password `

as for the current iface `permit`, it's just inline its current function body into `trust` and `password`

File: init/index.js [hunk 3]
```diff
@@ -22,9 +22,8 @@ async function init (link, dir, { ipc, header, autosubmit, defaults, force = fal
   let params = null
   if (isPear) {
     const { drive } = parseLink(link)
-    const trusted = new Set(await ipc.getPreference({ key: 'trusted' }) || [])
-    hypercoreid.encode(drive.key, hypercoreid.encode(drive.key), trusted)
-    if (trusted.has(hypercoreid.encode(drive.key)) === false) {
+    const trusted = await ipc.trusted()
```
Comment (davidmarkclements) [inline init/index.js:25]: let's do this: if (await ipc.trusted(drive.key) === false) throw...

File: cmd/init.js (hunk 2 above)
Comment (rafapaezbas) [reply cmd/init.js:57]: the issue is that `trust` and `password` all show different messages for stage, dump, run, init and info. Thats why we pass `explain`, `ask` and `act` as arguments.

File: cmd/init.js (hunk 1 above)
Comment (rafapaezbas) [reply cmd/init.js:48]: updated

File: cmd/init.js (hunk 2 above)
Comment (davidmarkclements) [reply cmd/init.js:54]: @rafapaezbas right right- so permit({ dialog: 'stage', .... })

and then the permit function owns all the output strings, so it passes the correct explain, ask, act args to trust and password

Comment (rafapaezbas) [reply cmd/init.js:47]: done

Comment (davidmarkclements) [reply cmd/init.js:47]: @rafapaezbas 
> this code from 45-57 here should just be one line

you're still not calling permit directly?

Comment (rafapaezbas) [reply cmd/init.js]: done

**Pull request merged**